test/parallel/test-net-server-simultaneous-accepts-produce-warning-once.js

```diff
@@ -1,12 +1,15 @@
 'use strict';
 
-const { expectWarning } = require('../common');
+const { expectWarning, mustCall, mustNotCall } = require('../common');
 const net = require('net');
 
 expectWarning(
   'DeprecationWarning',
   'net._setSimultaneousAccepts() is deprecated and will be removed.',
   'DEP0121');
 
+process.on('warning', mustCall(() => {
+  process.on('warning', mustNotCall());
+}));
```

Comment:
nit: this could be simplified as `mustCall` will check that the callback is only invoked once. We can also explicitly provide the number of calls we are expecting so it's clearer:
```suggestion
process.on('warning', mustCall(1));
```

Reply:
@aduh95 Your suggestion is simpler, but is slightly more permissive than the code in this PR. The code in this PR guarantees that the first call of the API results in a warning and the second call does not. The change you suggest here will still result in a passing test if the first call does not result in a warning and the second one does. Admittedly, that's an edge case that is unlikely to happen. But I have a slight preference for the stricter checking here.

Reply:
I don’t see how, the current code checks the `'warning'` event is emitted once, but I don’t think it can know what API call triggered it either 🤔

Reply:
> I don’t see how, the current code checks the `'warning'` event is emitted once, but I don’t think it can know what API call triggered it either 🤔

Current code in master branch tests it by having two separate tests that are nearly identical. One test only calls the API once and makes sure that the warning is emitted once. The other test calls the API twice and makes sure that the warning is emitted once. Alone, that second test doesn't confirm that the warning happens on the first API call, but in combination with the first test, that is effectively checked.

This PR consolidates those two tests into a single test.

Reply:
Oh, wait, or are you suggesting that there will be a problem if something *else* (like `--pending-deprecations`) causes warnings? Yeah, that's a valid concern and one we should address....

I'm wondering if we're actually best off just having two tests. :-D

Reply:
Two ways to consolidate the tests:

* Replace both tests with a test that launches a child_process and takes a parameter that tells the process how many times to call the API. No matter how many times, though, it should confirm that the warning happens just once. Then have it launch the process twice.
* Remove `expectsWarning()` and replace it with code that checks the warning within the file, basically consolidating the `process.on('warning', ...)` stuff in this PR with the `expectsWarning()` helper function functionality.

I do kinda think that just having two separate tests is easier. Maybe they each need a comment pointing to each other, though and explaining why they are nearly the same?

Reply:
I agree, keeping both tests seems better.

Reply:
@kuriyosh How would you feel about restoring both tests and adding a short comment in each one explaining why there are two different-but-very-similar tests?

Reply:
In my understanding, the discussed concern is that this test fails if other warnings occur in a second call. (Please point me out if I'm wrong here.) I agree that we need to run `test-net-deprecated-setsimultaneousaccepts.js` to make sure that `_setSimultaneousAccepts()` does not emit other warnings.

On the other hand, `test-net-server-simultaneous-accepts-produce-warning-once.js` in master branch does not check whether if `_setSimultaneousAccepts()` emit `DEP0121` in a second call. That test can be passed if second calling `_setSimultaneousAccepts()` emit `DEP0121` because of the behavior of [expectsWarning()](https://github.com/nodejs/node/blob/master/test/common/index.js#L569).
So, I think the change of `test-net-server-simultaneous-accepts-produce-warning-once.js` in this PR is needed.

Reply:
> On the other hand, `test-net-server-simultaneous-accepts-produce-warning-once.js` in master branch does not check whether if `_setSimultaneousAccepts()` emit `DEP0121` in a second call. 

It does because `expectsWarning()` will fail if it receives the warning more than once.

This exits with a failure status:

```js
const {expectWarning} = require('./test/common');

expectWarning('foo', 'foo', 'foo');
```

This exist with success status:

```javascript
const {expectWarning} = require('./test/common');

expectWarning('foo', 'foo', 'foo');

process.emitWarning('foo', 'foo', 'foo');
```

And this exist with a failure status again:

```javascript
const {expectWarning} = require('./test/common');

expectWarning('foo', 'foo', 'foo');

process.emitWarning('foo', 'foo', 'foo');
process.emitWarning('foo', 'foo', 'foo');
```

One thing I notice while running these tests is that the error message is not great on that last example because `const [ message, code ] = expected.shift();` (in `test/common/index.js`) does not accommodate the possibility that `expected` is an empty array and `expected.shift()` returns `undefined`:

```
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

So that's probably worth fixing.

Reply:
> So that's probably worth fixing.

https://github.com/nodejs/node/pull/41326

Reply:
Sorry, I misunderstood the behavior of `expectWarning`...
I totally agree on having two separate tests.
I have committed for adding comment in these test codes.